src/objective/regression_obj.cu

```diff
@@ -101,7 +108,7 @@ class RegLossObj : public ObjFunction {
 
           for (size_t idx = begin; idx < end; ++idx) {
             bst_float p = Loss::PredTransform(preds_ptr[idx]);
-            bst_float w = _is_null_weight ? 1.0f : weights_ptr[idx];
+            bst_float w = _is_null_weight ? 1.0f : weights_ptr[idx % n_targets];
```

Comment:
Are weights now per-target? Previously, weights are per-instance.

Reply:
Will fix it.

Reply:
Thanks for the catch.